astro/src/content/docs/operate/secure-and-monitor/elastic.mdx

```diff
@@ -0,0 +1,540 @@
+  time=$(echo "$time" | tr -d ' "')                      # 2024-06-21T05:14:56.123
+  time=$(echo "$time" | tr 'T' ' ')                      # 2024-06-21 05:14:56.123
+  sec="$(date -d "$(echo $time | cut -d '.' -f 1)" +%s)" # 1718946896
+  ms="$(echo $time | cut -d '.' -f 2)"                   # 123
+
+  # make the POST data
+  data=$(cat <<EOF
+{
+  "timestamp": ${sec}${ms},
+  "value": "${userId}"
+}
+EOF
+)
+
+  # send data to Elastic API
+  curl -X POST "${elasticUrl}/falogins/_doc" -H "Authorization: ApiKey ${elasticKey}" -H "Content-Type: application/json" -d "$data"
+
+done
+```
+
+This script gets all login records in the last hour to be sure the zip file has some data. In reality, replace 3600000 with 10000 so that when the script runs every ten seconds, it gets only the latest records. Note that FusionAuth uses milliseconds instead of the epoch standard of seconds, so the script has to append `000` to the normal Unix time.
+
+The file returned from FusionAuth unzips to `login_records.csv`, which looks like the data below.
+
+| "User Id "                           | "Time "                 | "Application Id "                    | "IP Address " | "City " | "Country " | "Zipcode " | "Region " | "Latitude " | "Longitude " |
+|--------------------------------------|-------------------------|--------------------------------------|---------------|---------|------------|------------|-----------|-------------|--------------|
+| ba81f3e2-3b0f-4d64-930f-38298de9dc0d | 2024-06-21T05:14:56.123 | 3c219e58-ed0e-4b18-ad48-f4f92793ae32 | 172.20.0.1    |         |            |            |           |             |              |
+| ba81f3e2-3b0f-4d64-930f-38298de9dc0d | 2024-06-21T05:07:06.406 | 3c219e58-ed0e-4b18-ad48-f4f92793ae32 | 172.20.0.1    |         |            |            |           |             |              |
+
+The records in this file look different from those in the FusionAuth console. Only Ids are given here, not email addresses or application names.
+
+The second half of the script reads in the CSV file, discards the header, and sends the user Id and the time of each login to Elastic.
+
+Create a file called `metricDockerfile`. Insert the content below.
+
+```
+FROM --platform=linux/amd64 alpine:3.19
+RUN apk add --no-cache curl jq nano
+COPY app.sh /app.sh
+RUN chmod +x app.sh
+CMD watch -t -n 10 /app.sh
+```
+
+Build the container with the command below.
+
+```sh
+docker build -f metricDockerfile --platform linux/amd64 -t metricimage .
+```
+
+Edit your `docker-compose.yml` file and add the new service as follows.
+
+```
+  fametric:
+    image: metricimage
+    container_name: fametric
+    networks:
+      - db_net
+```
+
+Now run all the containers with `docker compose up`.
+
+The output should be as below. Since you logged in in the last hour, there will be one row in the exported file and one value will be sent to Elastic. If you don't see any rows displayed on the screen, log in to FusionAuth again.
```

Comment:
```suggestion
The output should be as below. Since you logged-in in the last hour, there will be one row in the exported file and one value will be sent to Elastic. If you don't see any rows displayed on the screen, log in to FusionAuth again.
```

Reply:
the style guide asks for logged in 

[Use `logged in` instead of `logged-in`](https://github.com/FusionAuth/fusionauth-site/blob/449ff480b4c76075a37ecad855d2b5cd3f8b4792/DocsDevREADME.md?plain=1#L20)

Reply:
Sorry thought I pulled that comment back out, no need to update, the double "in in" is just a little weird.

Reply:
sorry missed that, totally agree. Changed the second one to within.

Reply:
@sixhobbits could you update the vale file, then I think you are good!